src/components/Routing.js

```diff
@@ -151,12 +158,23 @@ class Routing extends Component {
         return (
             <Container className='App' showBanner={(IS_MAINNET && !this.props.account.accountId) || !IS_MAINNET}>
                 <GlobalStyle />
-                <ConnectedRouter basename={PATH_PREFIX}  history={this.props.history}>
+                <ConnectedRouter basename={PATH_PREFIX} history={this.props.history}>
                     <ThemeProvider theme={theme}>
                         <ScrollToTop/>
                         <NetworkBanner accountId={this.props.account.accountId}/>
                         <Navigation/>
                         <GlobalAlert/>
+                        { 
+                            this.props.account.requestPending !== null &&
+                            <TwoFactorVerifyModal
+                                onClose={(verified) => {
+                                    const { account, promptTwoFactor } = this.props
+                                    // requestPending is Promise, resolve with verified from how Modal closed, verified or closed by user
```

Comment:
Not sure why we have this comment and then ignore what `requestPending` returns. Is this intentional that we don't wait for promise to resolve before calling `promptTwoFactor`?

Reply:
@mattlockyer

Reply:
updated comment to reflect more accurately what is happening